mobile/android/app/src/main/java/im/mustang/capa/nodeprocess/FileOperations.kt

```diff
@@ -0,0 +1,77 @@
+package im.mustang.capa.nodeprocess
+
+import android.content.res.AssetManager
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.withContext
+import java.io.File
+import java.io.FileOutputStream
+
+object FileOperations {
+    private const val TAG = "FileOperations"
+
+    // Asynchronous copying
+    suspend fun copyAssetsDir(assetManager: AssetManager, from: String, to: String): Boolean = coroutineScope {
+        try {
+            // Get the list of files in the assets directory
+            // if there's no files return
```

Comment:
```suggestion
            // if there's no files, then just return
```

Reply:
Why not throw an exception? The calling code throws anyways, if this returns false.